features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapReceiverSnmp4jImpl.java

```diff
@@ -116,53 +126,84 @@ public void setSyslogConnectionHandlers(TrapNotificationHandler handler) {
         m_trapNotificationHandlers = Collections.singletonList(handler);
     }
 
-    @Override
-    public String getName() {
-        String listenAddress = (m_config.getListenAddress() != null && m_config.getListenAddress().length() > 0) ? m_config.getListenAddress() : "0.0.0.0";
-        return getClass().getSimpleName() + " [" + listenAddress + ":" + m_config.getSyslogPort() + "]";
-    }
-
-    /*
-     * stop the current receiver
-     * @throws InterruptedException
-     * 
-     */
-    @Override
-    public void stop() throws InterruptedException {
-        m_stop = true;
-
-        // Close the datagram socket
-        if (m_dgSock != null) {
-            m_dgSock.close();
-        }
-
-        // Shut down the thread pools that are executing SyslogConnection and SyslogProcessor tasks
-        m_executor.shutdown();
-
-        if (m_context != null) {
-            LOG.debug("Stopping and joining thread context {}", m_context.getName());
-            m_context.interrupt();
-            m_context.join();
-            LOG.debug("Thread context stopped and joined");
+	@Override
+	public void trapReceived(TrapNotification trapNotification) {
+		try {
+			  for (TrapNotificationHandler handler : m_trapNotificationHandlers) {
+			    handler.handleTrapNotification(trapNotification);
+			  }
+			} catch (Throwable e) {
+			  LOG.error("Handler execution failed in {}", this.getClass().getSimpleName(), e);
+			}
+	}
+
+	@Override
+	public void trapError(int error, String msg) {
+      LOG.warn("Error Processing Received Trap: error = {} {}", error, (msg != null ? ", ref = " + msg : ""));
+	}
+	
+	@Override
+	public void start(){
+        try {
+        	InetAddress address = getInetAddress();
+        	LOG.info("Listening on {}:{}", address == null ? "[all interfaces]" : InetAddressUtils.str(address), m_snmpTrapPort);
+            SnmpUtils.registerForTraps(this, this, address, m_snmpTrapPort, m_snmpV3Users); // Need to clarify 
```

Comment:
without implementing this TrapProcessorFactory I was getting a compilation error.